client/src/components/classesPage/ClassCreateView.js

```diff
@@ -77,10 +75,132 @@ function ClassCreateView(props) {
     plain_content: campaignData.plain_content // requires [unsubscribe]
   };
 
+  useEffect(() => {
+    console.log(campaignData);
+  }, [campaignData]);
+
+  // sendgrix axios instance
+  const sgAx = axios.create({
+    baseURL: 'https://api.sendgrid.com/v3',
+    headers: {
+      authorization: `Bearer ${process.env.REACT_APP_SENDGRID_API_KEY}`
+    }
+  });
+
+  const token = localStorage.getItem('accessToken');
+
+  const ax = axios.create({
+    // baseURL: 'https://refreshr.herokuapp.com' // production
+    baseURL: 'http://localhost:9000',
+    headers: {
+      authorization: `Bearer ${token}` // development
+    }
+  });
+
+  const sgDb = async refreshrs => {
+    let firstPass = true; // flag to ensure we don't submit the same list to sendgrid
+    let list; // to store list id
+    for (let refreshr of refreshrs) {
+      console.log('list:', list);
+      console.log('refreshr:', refreshr);
+      // create list
+      console.log(listData.classnameInput);
+      console.log(recipientData);
+      if (firstPass) {
+        let body = {
+          name: listData.classnameInput
+        };
+        let res = await sgAx.post('/contactdb/lists', body);
+        console.log('res:', res);
+        list = res.data.id;
+        firstPass = false; // so we don't try to create duplicate lists
+
+        // save class and list to db
+        const classRes = await ax.post('/classes', {
+          name: listData.classnameInput,
+          sg_list_id: list
+        });
+
+        console.log(classRes);
+        const classId = classRes.data.newClassID;
+
+        console.log(classId);
+
+        // create recipients and add to list
+        const students = []; // to keep track of student id's for insert into students_classes
+        for (let recipient of recipientData) {
+          // create sg recipient
+          console.log(recipient);
+          let recipient_id = await sgAx.post('/contactdb/recipients', [
+            recipient
+          ]);
```

Comment:
So an sgAx request for every single recipient?

Reply:
Yeah, I was thinking about that. It would probably be better to send an array with all the recipients. This worked fine with two or three recipients, but I never tested it with more than that.